src/esn.contact.libs/app/addressBook/contact-api-client.service.js

```diff
@@ -615,5 +620,39 @@ require('../app.constant.js');
         return response.headers('X-ESN-TASK-ID');
       });
     }
+
+    /**
+      * Get contact Avatar
+      * @param  {Object} payload   The contact Id
+      * @param  {Object} options   Includes:
+      *                               + addressBookId
+      *                               + addressbookName
+      *                               + contactId
+      * @return {Promise}          If success :  resolve avatar data
+      *                            If not reject error
+      */
+
+    function getAvatar(payload) {
+      return contactRestangularService
+        .one('contacts/' + payload.addressBookId + '/' + payload.addressbookName + '/' + payload.contactId)
```

Comment:
```suggestion

        .one('contacts')
        .one(payload.addressBookId)
        .one(payload.addressbookName)
        .one(payload.contactId)
```

Reply:
we can't mock RestAngular with this way , it always generate error when using .one () .one ()...